### PR TITLE
Add validation to ensure numeric fields are not negative

### PR DESCRIPTION
**Pull request by djohnston32**
I rule and have completed some work on Case Manager that's ready for review!

Adds validation on numeric fields to ensure negative amounts of money aren't allowed.

Adds validation to the following fields:
* fulfillments
    * fund_payout
* patients
    * fund_pledge
    * naf_pledge
    * patient_contribution
    * procedure_cost
* archived_patients
    * fund_pledge
    * naf_pledge
    * patient_contribution
    * procedure_cost
* external_pledges
    * amount
* events
    * pledge_amount

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: app/models/archived_patient.rb [hunk 1]
```diff
@@ -35,6 +35,11 @@ class ArchivedPatient < ApplicationRecord
             presence: true
   validates :appointment_date, format: /\A\d{4}-\d{1,2}-\d{1,2}\z/,
                                allow_blank: true
+  validates :procedure_cost, numericality: { only_integer: true, allow_nil: true, greater_than: 0 }
```
Comment (colinxfleming) [inline app/models/archived_patient.rb:38]: my experience with accountants is that they looooove being able to zero things out as an indicator to themselves of something or other. I think for every field in all these models we can validate `only_integer: true, greater_than_or_equal_to: 0` like you are for the other fields here.

**Review by colinxfleming — CHANGES_REQUESTED** [review]
This is right on. I think seeing this in action, I'd like to slightly expand the scope of this - there's not really a reason to not have most of our int fields be zero or up, right?

I think two changes here:

* validate `greater_than_or_equal_to: 0` on everything, rather than `greater_than: 0`
* expand this to the following integer fields on the following models:

```
Clinic: costs_5wks thru costs_30wks, gestational_limit
fulfillment: gestation_at_procedure
patient: last menstrual period weeks, last menstrual period days, age, household size children, household size adults
```

sorry for the change in direction!

File: app/models/archived_patient.rb (hunk 1 above)
Comment (djohnston32) [reply app/models/archived_patient.rb]: Sounds good! Updated that and added the rest of the fields in the latest revision.

**Review by colinxfleming — APPROVED** [review]
thank you very much, and thank you again for handling the change in direction so well!

**Pull request merged**